File: src/01-easy/temperatures/solution.hs
import System.IO
main :: IO ()
main = do
    hSetBuffering stdout NoBuffering
    n <- readLn :: IO Int
    if n == 0 then print 0
    else do
        line <- getLine
        let t = map read (words line) :: [Int]
            closer a b = if abs a < abs b || (abs a == abs b && a > 0) then a else b
        print (foldl1 closer t)

File: src/01-easy/temperatures/solution.pas
{$H+}
program Answer;
var
    n, i, v, r: integer;
begin
    readln(n);
    if n = 0 then begin writeln(0); halt; end;
    r := 99999;
    for i := 1 to n do
    begin
        read(v);
        if (abs(v) < abs(r)) or ((abs(v) = abs(r)) and (v > 0)) then r := v;
    end;
    writeln(r);
    flush(output);
end.
